# 01 Importing Libraries

In [2]:
# Import libraries
import pandas as pd
import numpy as np
import os

# 02 Import Data

In [4]:
path = r'C:\Users\isava\OneDrive\Documents\CareerFoundry\Data Immersion\PythonFundamentals\Instacart Basket Analysis'
path

'C:\\Users\\isava\\OneDrive\\Documents\\CareerFoundry\\Data Immersion\\PythonFundamentals\\Instacart Basket Analysis'

In [5]:
df_ords_prods = pd.read_pickle(os.path.join(path,'02 Data', 'Prepared Data','orders_products_combined.pkl'))
df_prods = pd.read_csv(os.path.join(path,'02 Data', 'Prepared Data','products_checked.csv'), index_col = False)

In [6]:
df_ords_prods.shape

(32434489, 10)

### Save Memory

In [8]:
df_prods.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49672 entries, 0 to 49671
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Unnamed: 0     49672 non-null  int64  
 1   product_id     49672 non-null  int64  
 2   product_name   49672 non-null  object 
 3   aisle_id       49672 non-null  int64  
 4   department_id  49672 non-null  int64  
 5   prices         49672 non-null  float64
dtypes: float64(1), int64(4), object(1)
memory usage: 2.3+ MB


In [9]:
# Further columns from other data sets
df_prods.drop(columns = {'Unnamed: 0'}, inplace = True)

In [10]:
df_prods['aisle_id'] = df_prods['aisle_id'].astype('int8')

In [11]:
df_prods['department_id'] = df_prods['department_id'].astype('int8')
#df_prods['prices'] = df_prods['prices'].astype('float16') this creates inf variables that we dont want

In [12]:
df_prods.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49672 entries, 0 to 49671
Data columns (total 5 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   product_id     49672 non-null  int64  
 1   product_name   49672 non-null  object 
 2   aisle_id       49672 non-null  int8   
 3   department_id  49672 non-null  int8   
 4   prices         49672 non-null  float64
dtypes: float64(1), int64(1), int8(2), object(1)
memory usage: 1.2+ MB


# 03 Merging Data

In [14]:
df_ords_prods.head()

,order_id,user_id,order_number,orders_day_of_week,order_time_of_day,days_since_users_prior_order,product_id,add_to_cart_order,reordered,_merge
0,2539329,1,1,2,8,NaN,196,1,0,both
1,2539329,1,1,2,8,NaN,14084,2,0,both
2,2539329,1,1,2,8,NaN,12427,3,0,both
3,2539329,1,1,2,8,NaN,26088,4,0,both
4,2539329,1,1,2,8,NaN,26405,5,0,both


In [15]:
df_prods.head()

,product_id,product_name,aisle_id,department_id,prices
0,1,Chocolate Sandwich Cookies,61,19,5.8
1,2,All-Seasons Salt,104,13,9.3
2,3,Robust Golden Unsweetened Oolong Tea,94,7,4.5
3,4,Smart Ones Classic Favorites Mini Rigatoni Wit...,38,1,10.5
4,5,Green Chile Anytime Sauce,5,13,4.3


We can merge on product id.

In [17]:
df_merged_large = df_ords_prods.merge(df_prods, on = 'product_id', indicator = '_prodMerge') #Indicator throwing error because there cannot be two columns names _Merge so I created a custom one here

In [18]:
df_merged_large['_prodMerge'].value_counts()

_prodMerge
both          32404859
left_only            0
right_only           0
Name: count, dtype: int64

The resulting dataframe has 32,404,859 rows.

# 04 Export Data Frame

In [21]:
#df_merged_large.to_csv(os.path.join(path, '02 Data','Prepared Data', 'orders_products_combined.csv'))
df_merged_large.to_pickle(os.path.join(path, '02 Data','Prepared Data', 'ords_prods_merge.pkl'))